indica/readers/adas.py

```diff
@@ -183,7 +186,175 @@ def get_adf15(
+        file_component = build_file_component(year, element)
+        filename = Path(pathname2url(file_component)) / pathname2url(
+            f"{file_component}_{filetype.lower()}]"
+            f"[{element.lower()}{charge.lower()}.dat"
+        )
+
+        header_match = {
+            "compact": r"(\d+).+/(\S+).*\+(.*)photon",
+            "expanded": r"(\d+).+/(\S+).*\:(.*)photon",
+        }
+        section_header_match = {
+            "compact": r"(\d+.\d+).+\s+(\d+)\s+(\d+).+type\s?"
+            r"=\s?(\S+).+isel.+\s+(\d+)",
+            "expanded": r"(\d+.\d+)\s+(\d+)\s+(\d+).+type\s?="
+            r"\s?(\S+).+isel\s+?=\s+?(\d+)",
+        }
+        with self._get_file("adf15", filename) as f:
+            header = f.readline().strip().lower()
+            identifier = file_type(header.split("/")[1][2])
+
+            match = header_match[identifier]
+            m = re.search(match, header, re.I)
+            assert isinstance(m, re.Match)
+            ntrans = int(m.group(1))
+            element_name = m.group(2).strip().lower()
+            charge_state = int(m.group(3))
+            assert element_name == element.lower()
+            m = re.search(r"(\d+)(\S*)", charge)
+            assert isinstance(m, re.Match)
+            extracted_charge = m.group(1)
+            if charge_state != int(extracted_charge):
+                raise ValueError(
+                    f"Charge state in ADF15 file ({charge_state}) does not "
+                    f"match argument ({charge})."
+                )
+
+            # Read first section header to build arrays outside of reading loop
+            match = section_header_match[identifier]
+            header_re = re.compile(match)
+            m = None
+            while not m:
+                line = f.readline().strip().lower()
+                if not line:
+                    break
+                m = header_re.search(line)
+            assert isinstance(m, re.Match)
+            nd = int(m.group(2))
+            nt = int(m.group(3))
+            data = np.empty((ntrans, nt, nd))
+            ttype: List[str] = []
+            tindex = np.empty(ntrans)
+            wavelength = np.empty(ntrans)
+
+            # Read Photon Emissivity Coefficient rates
+            for i in range(ntrans):
+                m = header_re.search(line)
+                assert isinstance(m, re.Match)
+                assert int(m.group(5)) - 1 == i
+                tindex[i] = i + 1
+                ttype.append(m.group(4))
+                wavelength[i] = float(m.group(1))  # (Angstroms)
+                densities = np.fromfile(f, float, nd, " ")
+                temperatures = np.fromfile(f, float, nt, " ")
+                data[i, ...] = np.fromfile(f, float, nd * nt, " ").reshape((nt, nd))
+                line = f.readline().strip().lower()
+
+            # Read Transition information from end of file
+            file_end_re = re.compile(r"c\s+[isel].+\s+[transition].+\s+[type]")
```

Comment:
Regex is matching a blank line for the Ar10 test case, causes the transition_match later to fail therefore all tests for Ar10 are failing

Reply:
Fixed this: it was down to the block **if not line:** which was making it crash.

Still need to set a working EOF block in these while loops. If you have recommendations, I'm open to suggestions!